pkg/render/node.go

```diff
@@ -59,14 +71,21 @@ func Node(
 	aci *operator.AmazonCloudIntegration,
 	migrate bool,
 ) Component {
-	return &nodeComponent{
+	node := &nodeComponent{
 		cr:              cr,
 		netConfig:       nc,
 		birdTemplates:   bt,
 		typhaNodeTLS:    tnTLS,
 		amazonCloudInt:  aci,
 		migrationNeeded: migrate,
 	}
+
+	if k8sHost, k8sPort, err := getK8sEndpoint(); err == nil {
```

Comment:
I'm not a fan of adding this dependency on the KUBERNETES_SERVICE_HOST / PORT environment variables to the render code (and really the operator in general), since it's totally valid for them to not be present in some scenarios. 

As an example, a pretty standard way of testing the operator code during development is something like this:

```
KUBECONFIG=./kubeconfig.yaml go run ./cmd/manager
```

Could we use the auto-configured kube-client in order to extract the configured API endpoint and pass it to render instead? Would that work?

Reply:
Maybe me you and @tmjd should hop on a call to get alignment on what we want to do here.

Reply:
> As an example, a pretty standard way of testing the operator code during development is something like this:
> 
> ```
> KUBECONFIG=./kubeconfig.yaml go run ./cmd/manager
> ```

That scenario works. You need to set all 3. KUBECONFIG takes precedence and KUBERNETES_SERVICE_HOST/PORT are just forwarded. If it is in-cluster, KUBECONFIG (afaict) is not set and thus the operator's client uses the override as well.

Reply:
My point is that we previously did not have a dependency on the KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT environment variables.

If you ran the code without them set, then it still worked. With this change, if those env vars are not set, then the code will error. I'm not yet convinced that's an OK change. Does that make sense?

Reply:
> If it is in-cluster, KUBECONFIG (afaict) is not set and thus the operator's client uses the override as well.

It would totally be possible for someone to:
- Mount in a kubeconfig file
- Set the KUBECONFIG environment variable. 

With this change, that workflow is no longer possible (not that I'm aware of anyone using it that way necessarily, but trying to show how this is a breaking change).

Reply:
> If you ran the code without them set, then it still worked. With this change, if those env vars are not set, then the code will error. I'm not yet convinced that's an OK change. Does that make sense?

Then the forward values are not set and felix may not work correctly if it cannot reach k8s. In a cluster with kube-proxy that is OK as felix will use whatever values it has set. In a cluster without kube-proxy, e.g. when enabling bpf, operator will not be able to connect to k8s without this being set properly. That is a requirement.

We can revert to using the OVERRIDE vars. Is that what you suggest?